lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -85,55 +134,14 @@ private void Initialize()
         /// <returns></returns>
         public T GetPartitionHandle()
         {
-            if (_copyToLocal)
-            {
-                if (!_isInitialized)
-                {
-                    Initialize();
-                }
-
-                return _fileSerializer.Deserialize(_localFiles);
-            }
-            return _fileSerializer.Deserialize(_filePaths);
-        }
-
-        private void CopyFromRemote()
-        {
-            string localFileFolder = _tempFileCreator.CreateTempDirectory("-partition-");
-            Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "Local file temp folder: {0}", localFileFolder));
-
-            foreach (var sourceFilePath in _filePaths)
+            lock (_lock)
             {
-                Uri sourceUri = _fileSystem.CreateUriForPath(sourceFilePath);
-                Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "sourceUri {0}: ", sourceUri));
-                if (!_fileSystem.Exists(sourceUri))
+                if (!_localFiles.IsPresent())
```

Comment:
This back to the JIra discussion today.  What does Cache mean? It is to download files from remote to local, or it is to cache the data into memory? 

If it is only to download the data from remote to local, basically it is to change the original Initialize to public with rename and minor modification. If it is to load the data to memory, do we have a separate Jira?